test/core/symbol/MerkleHashBuilder.spec.ts

```diff
@@ -0,0 +1,132 @@
+    });
+
+    it('Can build from blanced tree', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder()
+            .update(Converter.hexToUint8('215b158f0bd416b596271bce527cd9dc8e4a639cc271d896f9156af6f441eeb9'))
+            .update(Converter.hexToUint8('976c5ce6bf3f797113e5a3a094c7801c885daf783c50563ffd3ca6a5ef580e25'));
+
+        // Act:
+        const rootHash = builder.final();
+
+        // Assert:
+        expect(Converter.uint8ToHex(rootHash).toLocaleLowerCase()).equal(
+            '1c704e3ac99b124f92d2648649ec72c7a19ea4e2bb24f669b976180a295876fa',
+        );
+    });
+
+    it('Can build from unblanced tree', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder()
+            .update(Converter.hexToUint8('215b158f0bd416b596271bce527cd9dc8e4a639cc271d896f9156af6f441eeb9'))
+            .update(Converter.hexToUint8('976c5ce6bf3f797113e5a3a094c7801c885daf783c50563ffd3ca6a5ef580e25'))
+            .update(Converter.hexToUint8('e926cc323886d47234bb0b49219c81e280e8a65748b437c2ae83b09b37a5aaf2'));
+
+        // Act:
+        const rootHash = builder.final();
+
+        // Assert:
+        expect(Converter.uint8ToHex(rootHash).toLocaleLowerCase()).equal(
+            '5dc17b2409d50bcc7c1faa720d0ec8b79a1705d0c517bcc0bdbd316540974d5e',
+        );
+    });
+
+    it('Can build from unblanced tree', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder()
+            .update(Converter.hexToUint8('215b158f0bd416b596271bce527cd9dc8e4a639cc271d896f9156af6f441eeb9'))
+            .update(Converter.hexToUint8('976c5ce6bf3f797113e5a3a094c7801c885daf783c50563ffd3ca6a5ef580e25'))
+            .update(Converter.hexToUint8('e926cc323886d47234bb0b49219c81e280e8a65748b437c2ae83b09b37a5aaf2'));
+
+        // Act:
+        const rootHash = builder.final();
+
+        // Assert:
+        expect(Converter.uint8ToHex(rootHash).toLocaleLowerCase()).equal(
+            '5dc17b2409d50bcc7c1faa720d0ec8b79a1705d0c517bcc0bdbd316540974d5e',
+        );
+    });
+
+    it('Change sub hash order changes merkle hash', () => {
+        // Arrange:
+        const hash1 = Array.from({ length: 8 }, () => Crypto.randomBytes(32));
+        const hash2: Buffer[] = [];
+        [0, 1, 2, 5, 4, 3, 6, 7].forEach((index) => hash2.push(hash1[index]));
+
+        // Sanity:
+        expect(hash1.length).equal(hash2.length);
+
+        // Act:
+        const merkle1 = calculateMerkleHash(hash1);
+        const merkle2 = calculateMerkleHash(hash2);
+
+        // Assert:
+        expect(merkle1).not.deep.equal(merkle2);
+    });
+
+    it('Change sub hash changes merkle hash', () => {
+        // Arrange:
+        const hash1 = Array.from({ length: 8 }, () => Crypto.randomBytes(32));
+        const hash2: Buffer[] = [];
+        [0, 1, 2, 3, -1, 5, 6, 7].forEach((index) => hash2.push(index <= 0 ? Crypto.randomBytes(32) : hash1[index]));
```

Comment:
not sure if your intent is to replace 2 (0, -1) hashes or 1?

Reply:
this is to replace 2(0, -1) to random hash bytes